### PR TITLE
docs: plugin.md add config notice for `addSingleton`  function

### PR DESCRIPTION
**Pull request by rccoder**
##### Checklist

- [x] documentation is changed or added
- [x] commit message follows commit guidelines

##### Description of change

Recent I try to develop a egg plugin for my project，and was troubled by `addSingleton` function just because config file need `client` or `clients` key configuration as the config to the `creator` function。

And it seems that many people have encountered the same problem before.

- https://cnodejs.org/topic/59a5232ad97b7e2308242846
- https://github.com/eggjs/egg/issues/552
- https://github.com/eggjs/egg/issues/2226
- https://github.com/eggjs/egg/issues/1955

### CODE REVIEW

File: docs/source/zh-cn/advanced/plugin.md [hunk 1]
```diff
@@ -269,7 +269,7 @@ $ npm test
 - 在一个应用中同时使用同一个服务的不同实例（连接到两个不同的 MySQL 数据库）。
 - 从其他服务获取配置后动态初始化连接（从配置中心获取到 MySQL 服务地址后再建立连接）。
 
-如果让插件各自实现，可能会出现各种奇怪的配置方式和初始化方式，所以框架提供了 `app.addSingleton(name, creator)` 方法来统一这一类服务的创建。
+如果让插件各自实现，可能会出现各种奇怪的配置方式和初始化方式，所以框架提供了 `app.addSingleton(name, creator)` 方法来统一这一类服务的创建。需要注意的是在使用 `app.addSingleton(name, creator)` 方法时，配置文件中一定要有 `client` 或者 `client` 为 key 的配置作为传入 `creator` 函数 的 `config`。
```
Comment (atian25) [inline docs/source/zh-cn/advanced/plugin.md]: - 后面那个 client 少 s 了
- 最好注明下常见的一个问题，就是配置了 clients ，但是用的是 this.redis.xx() 导致 undefined

Comment (rccoder) [reply docs/source/zh-cn/advanced/plugin.md]: @atian25  第二个没太明白，要表达的意思是应该从 app 上取 ？

Comment (atian25) [reply docs/source/zh-cn/advanced/plugin.md]: 这块开发者经常遇到的一个问题是：
- 如果配置了 `client`，就要 `app.mysql.query()`
- 如果配置了 `clients`，就要 `app.mysql.get('instanceId').query()`

很多人配置了 clients 却用第一种的方式，结果报错 app.mysql.query() undefiend

Comment (rccoder) [reply docs/source/zh-cn/advanced/plugin.md]: @atian25  在 多实例 一节中加了注意点

**Pull request merged**